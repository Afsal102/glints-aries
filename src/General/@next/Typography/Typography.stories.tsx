import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Typography, TypographyProps } from './Typography';
import { BaseContainer } from '../../../Layout/GlintsContainer/GlintsContainer';
import * as Colors from '../../../Utils/@next/colors';

Typography.displayName = 'Typography.Paragraph';

export default {
  title: 'Typography/@next/Typography',
  component: Typography,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
  argTypes: {
    color: { control: 'color' },
  },
} as Meta;

const defaultText =
  "Tanya farnuva naar cam utinu en lokirim tanka tel' taurnin. Lirimaerea vedui' il'er Lotesse Shaalth.";

const Template: Story<TypographyProps> = args => (
  <Typography {...args}>{args.children || defaultText}</Typography>
);

export const Interactive = Template.bind({});
Interactive.args = {};

export const Color = Template.bind({});
Color.args = { color: Colors.Blue.Brand };

export const Ellipsis = Template.bind({});
Ellipsis.args = {
  ellipsis: true,
  children:
    "'Quel re amrun undome Beikaerea lle auta yeste' Mellonea. Guinahiroo ai' atar diola lle Mith'quessir. Tanya nae sai eina quel undome mani naa tanya nat' loomi en gurth. A! manke naa nae lle tinechor Tarion. Guinahiroo tanya awra! Heruamin Hallaerea. Mani uma lle merna ten' ta toror' taur'ohtarie Iire Cermie. Mankoi bragol alaa Morierea tal dagore.",
};

export const Bold = Template.bind({});
Bold.args = { bold: true };

export const Variant = Template.bind({});
Variant.args = { variant: 'body1' };
