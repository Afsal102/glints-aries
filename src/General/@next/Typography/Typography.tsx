import React, { ReactNode } from 'react';

import { StyledTypography } from './TypographyStyles';

export type Variant =
  | 'headline1'
  | 'headline2'
  | 'headline3'
  | 'headline4'
  | 'headline5'
  | 'headline6'
  | 'subtitle1'
  | 'subtitle2'
  | 'body1'
  | 'body2'
  | 'button'
  | 'caption'
  | 'overline';

export interface TypographyProps {
  as?: React.ElementType;
  children: ReactNode;
  variant?: Variant;
}

export const Typography = ({
  as,
  children,
  variant,
  ...props
}: TypographyProps) => {
  return (
    <StyledTypography as={as ? as : 'p'} variant={variant} {...props}>
      {children}
    </StyledTypography>
  );
};
