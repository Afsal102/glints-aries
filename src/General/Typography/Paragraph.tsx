import * as React from 'react';
import classNames from 'classnames';

import {
  Paragraph as StyledParagraph,
  ParagraphProps,
} from './ParagraphStyles';
import { SecondaryColor } from '../../Utils/Colors';

const Paragraph: React.FunctionComponent<Props> = props => {
  const {
    className,
    children,
    variant = 'regular',
    bold,
    color = SecondaryColor.black,
    ellipsis,
    ...restProps
  } = props;

  return (
    <StyledParagraph
      className={classNames('aries-typography-paragraph', className)}
      variant={variant}
      bold={bold}
      color={color}
      ellipsis={ellipsis}
      {...restProps}
    >
      {children}
    </StyledParagraph>
  );
};

interface Props extends ParagraphProps {
  className?: string;
  children: React.ReactNode;
}

export default Paragraph;
