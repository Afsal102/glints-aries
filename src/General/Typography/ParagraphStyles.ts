import styled from 'styled-components';

import { SecondaryColor } from '../../Style/Colors';

export interface ParagraphProps {
  bold?: boolean;
  color?: string;
  ellipsis?: boolean;
  variant?: 'subtitle' | 'regular' | 'caption' | 'smallest';
}

export const Paragraph = styled.p<ParagraphProps>`
  margin: 0;
  font-weight: ${props => (props.bold ? 'bold' : 'normal')};
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  color: ${props => props.color || SecondaryColor.black};

  ${props => {
    switch (props.variant) {
      case 'subtitle':
        return `
          font-size: 18px;
        `;
      case 'regular':
        return `
          font-size: 16px;
        `;
      case 'caption':
        return `
          font-size: 14px;
        `;
      case 'smallest':
        return `
          font-size: 12px;
        `;
      default:
        return `
          font-size: 16px;
        `;
    }
  }};

  ${props => {
    if (props.ellipsis) {
      return `
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
      `;
    }
  }}
`;
