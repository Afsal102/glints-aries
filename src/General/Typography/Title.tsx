import * as React from 'react';
import classNames from 'classnames';

import { Title as StyledTitle } from './TitleStyles';
import { SecondaryColor } from '../../Style/Colors';

const Title: React.FunctionComponent<Props> = props => {
  const {
    className,
    children,
    color = SecondaryColor.black,
    ellipsis,
    tag = 'h1',
    ...restProps
  } = props;

  return (
    <StyledTitle
      className={classNames('aries-typography-title', className)}
      as={tag}
      color={color}
      ellipsis={ellipsis}
      {...restProps}
    >
      {children}
    </StyledTitle>
  );
};

export type tagType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

interface Props {
  className?: string;
  children: React.ReactNode;
  color?: string;
  ellipsis?: boolean;
  tag?: tagType;
}

export default Title;
