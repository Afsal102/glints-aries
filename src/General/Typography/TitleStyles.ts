import styled from 'styled-components';

import { tagType } from './Title';

export const TITLE_VARIANTS = {
  h1: 'h1',
  h2: 'h2',
  h3: 'h3',
  h4: 'h4',
  h5: 'h5',
  h6: 'h6',
};

export const TITLE_FONT_SIZES = {
  [TITLE_VARIANTS.h1]: 50,
  [TITLE_VARIANTS.h2]: 36,
  [TITLE_VARIANTS.h3]: 30,
  [TITLE_VARIANTS.h4]: 26,
  [TITLE_VARIANTS.h5]: 22,
  [TITLE_VARIANTS.h6]: 18,
};

interface TitleProps {
  as: tagType;
  color?: string;
  ellipsis?: boolean;
}

export const Title = styled.h1<TitleProps>`
  margin: 0;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  letter-spacing: normal;
  color: ${props => props.color};

  ${props => {
    switch (props.as) {
      case TITLE_VARIANTS.h1:
        return `
        font-size: ${TITLE_FONT_SIZES.h1}px;
        font-weight: 900;
        line-height: 1.2em;
      `;
      case TITLE_VARIANTS.h2:
        return `
        font-size: ${TITLE_FONT_SIZES.h2}px;
        line-height: 1.2em;
      `;
      case TITLE_VARIANTS.h3:
        return `
        font-size: ${TITLE_FONT_SIZES.h3}px;
        line-height: 1.4em;
      `;
      case TITLE_VARIANTS.h4:
        return `
        font-size: ${TITLE_FONT_SIZES.h4}px;
        line-height: 1.4em;
      `;
      case TITLE_VARIANTS.h5:
        return `
        font-size: ${TITLE_FONT_SIZES.h5}px;
        line-height: 1.4em;
      `;
      case TITLE_VARIANTS.h6:
        return `
        font-size: ${TITLE_FONT_SIZES.h6}px;
        line-height: 1.4em;
      `;
      default:
        return `
        font-size: ${TITLE_FONT_SIZES.h1}px;
        font-weight: 900;
        line-height: 1.2em;
    `;
    }
  }};

  ${props => {
    if (props.ellipsis) {
      return `
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
      `;
    }
  }}
`;
