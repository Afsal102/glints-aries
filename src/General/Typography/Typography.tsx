import * as React from 'react';

import Title from './Title';
import Paragraph from './Paragraph';

const Typography: Typography = props => {
  const { children } = props;

  return <article>{children}</article>;
};

type Typography = React.FunctionComponent<Props> & {
  Title: typeof Title;
  Paragraph: typeof Paragraph;
};

Typography.Title = Title;
Typography.Paragraph = Paragraph;

interface Props {
  children: React.ReactNode;
}

export default Typography;
