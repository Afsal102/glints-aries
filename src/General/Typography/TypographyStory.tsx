import * as React from 'react';

import StorybookComponent from '../../../stories/StorybookComponent';

import Typography from '.';
import { TITLE_VARIANTS, TITLE_FONT_SIZES } from './TitleStyles';
import { PARAGRAPH_VARIANTS, PARAGRAPH_FONT_SIZES } from './ParagraphStyles';
import Heading from '../Heading';
import { PrimaryColor } from '../../Style/Colors';

const { Title, Paragraph } = Typography;

const titleProps = {
  'Typography.Title': [
    {
      name: 'color',
      type: 'string',
      defaultValue: <code>black</code>,
      possibleValue: <code>hex-value | rba-value | string-value</code>,
      require: 'no',
      description: "Sets the Title's color.",
    },
    {
      name: 'ellipsis',
      type: 'boolean',
      defaultValue: <code>false</code>,
      possibleValue: <code>true | false</code>,
      require: 'no',
      description: 'Display ellipsis when the title overflows.',
    },
    {
      name: 'tag',
      type: 'string',
      defaultValue: <code>h1</code>,
      possibleValue: (
        <code>
          {TITLE_VARIANTS.h1} | {TITLE_VARIANTS.h2} | {TITLE_VARIANTS.h3} |{' '}
          {TITLE_VARIANTS.h4} | {TITLE_VARIANTS.h5} | {TITLE_VARIANTS.h6}
        </code>
      ),
      require: 'no',
      description: 'Sets the title tag to one of the <h1> - <h6> tags.',
    },
  ],
};

const paragraphProps = {
  'Typography.Paragraph': [
    {
      name: 'bold',
      type: 'boolean',
      defaultValue: <code>false</code>,
      possibleValue: <code>true | false</code>,
      require: 'no',
      description: "Sets the Paragraph's font-weight to bold.",
    },
    {
      name: 'color',
      type: 'string',
      defaultValue: <code>black</code>,
      possibleValue: <code>hex-value | rba-value | string-value</code>,
      require: 'no',
      description: "Sets the Paragraph's color.",
    },
    {
      name: 'ellipsis',
      type: 'boolean',
      defaultValue: <code>false</code>,
      possibleValue: <code>true | false</code>,
      require: 'no',
      description: 'Display ellipsis when the title overflows.',
    },
    {
      name: 'variant',
      type: 'string',
      defaultValue: <code>regular</code>,
      possibleValue: (
        <code>
          {PARAGRAPH_VARIANTS.subtitle} | {PARAGRAPH_VARIANTS.regular} |{' '}
          {PARAGRAPH_VARIANTS.caption} | {PARAGRAPH_VARIANTS.smallest}
        </code>
      ),
      require: 'no',
      description: "Sets the Paragraph's variant.",
    },
  ],
};

const TitleStory = () => (
  <StorybookComponent
    title="Typography"
    code="import { Typography } from 'glints-aries'"
    propsObject={titleProps}
    usage={`import { Typography, PrimaryColor } from 'glints-aries';

const { Title } = Typography;

ReactDOM.render(
  <div style={{ display: 'grid', gridRowGap: '15px' }}>
    <Title tag="h1">Heading 1</Title>
    <Title tag="h2">Heading 2</Title>
    <Title tag="h3">Heading 3</Title>
    <Title tag="h4">Heading 4</Title>
    <Title tag="h5">Heading 5</Title>
    <Title tag="h6">Heading 6</Title>
    <Title tag="h5" color={PrimaryColor.glintsred}>
      Colored Title
    </Title>
    <Title tag="h5" ellipsis>
      Title with Ellipses - As the first sign in the zodiac, the presence of
      Aries always marks the beginning of something energetic and turbulent.
      They are continuously looking for dynamic, speed and competition, always
      being the first in everything - from work to social gatherings.
    </Title>
  </div>,
  mountNode
);`}
  >
    <Heading style={{ fontSize: '24px', marginBottom: '1em' }}>
      Typography.Title
    </Heading>
    <div style={{ display: 'grid', gridRowGap: '15px' }}>
      <Title tag="h1">Heading 1 - {TITLE_FONT_SIZES.h1}px</Title>
      <Title tag="h2">Heading 2 - {TITLE_FONT_SIZES.h2}px</Title>
      <Title tag="h3">Heading 3 - {TITLE_FONT_SIZES.h3}px</Title>
      <Title tag="h4">Heading 4 - {TITLE_FONT_SIZES.h4}px</Title>
      <Title tag="h5">Heading 5 - {TITLE_FONT_SIZES.h5}px</Title>
      <Title tag="h6">Heading 6 - {TITLE_FONT_SIZES.h6}px</Title>
      <Title tag="h5" color={PrimaryColor.glintsred}>
        Colored Title
      </Title>
      <Title tag="h5" ellipsis>
        Title with Ellipses - As the first sign in the zodiac, the presence of
        Aries always marks the beginning of something energetic and turbulent.
        They are continuously looking for dynamic, speed and competition, always
        being the first in everything - from work to social gatherings.
      </Title>
    </div>
  </StorybookComponent>
);

const ParagraphStory = () => (
  <StorybookComponent
    propsObject={paragraphProps}
    usage={`import { Typography, PrimaryColor } from 'glints-aries';

const { Paragraph } = Typography;

ReactDOM.render(
  <div style={{ display: 'grid', gridRowGap: '15px' }}>
    <Paragraph variant="subtitle">
      Subtitle <br />
      As the first sign in the zodiac, the presence of Aries always marks the
      beginning of something energetic and turbulent.
    </Paragraph>
    <Paragraph variant="regular">
      Regular <br />
      As the first sign in the zodiac, the presence of Aries always marks the
      beginning of something energetic and turbulent.
    </Paragraph>
    <Paragraph variant="caption">
      Caption <br />
      As the first sign in the zodiac, the presence of Aries always marks the
      beginning of something energetic and turbulent.
    </Paragraph>
    <Paragraph variant="smallest">
      Smallest <br />
      As the first sign in the zodiac, the presence of Aries always marks the
      beginning of something energetic and turbulent.
    </Paragraph>
    <Paragraph bold>
      Bold Paragraph <br />
      As the first sign in the zodiac, the presence of Aries always marks the
      beginning of something energetic and turbulent.
    </Paragraph>
    <Paragraph color={PrimaryColor.glintsred}>
      Colored Paragraph <br />
      As the first sign in the zodiac, the presence of Aries always marks the
      beginning of something energetic and turbulent.
    </Paragraph>
    <Paragraph ellipsis>
      Paragraph with Ellipses - As the first sign in the zodiac, the presence
      of Aries always marks the beginning of something energetic and
      turbulent. They are continuously looking for dynamic, speed and
      competition, always being the first in everything - from work to social
      gatherings.
    </Paragraph>
  </div>,
  mountNode
);`}
  >
    <Heading style={{ fontSize: '24px', marginBottom: '1em' }}>
      Typography.Paragraph
    </Heading>
    <div style={{ display: 'grid', gridRowGap: '15px' }}>
      <Paragraph variant="subtitle">
        Subtitle - {PARAGRAPH_FONT_SIZES.subtitle}px
        <br />
        As the first sign in the zodiac, the presence of Aries always marks the
        beginning of something energetic and turbulent.
      </Paragraph>
      <Paragraph variant="regular">
        Regular - {PARAGRAPH_FONT_SIZES.regular}px
        <br />
        As the first sign in the zodiac, the presence of Aries always marks the
        beginning of something energetic and turbulent.
      </Paragraph>
      <Paragraph variant="caption">
        Caption - {PARAGRAPH_FONT_SIZES.caption}px
        <br />
        As the first sign in the zodiac, the presence of Aries always marks the
        beginning of something energetic and turbulent.
      </Paragraph>
      <Paragraph variant="smallest">
        Smallest - {PARAGRAPH_FONT_SIZES.smallest}px
        <br />
        As the first sign in the zodiac, the presence of Aries always marks the
        beginning of something energetic and turbulent.
      </Paragraph>
      <Paragraph bold>
        Bold Paragraph <br />
        As the first sign in the zodiac, the presence of Aries always marks the
        beginning of something energetic and turbulent.
      </Paragraph>
      <Paragraph color={PrimaryColor.glintsred}>
        Colored Paragraph <br />
        As the first sign in the zodiac, the presence of Aries always marks the
        beginning of something energetic and turbulent.
      </Paragraph>
      <Paragraph ellipsis>
        Paragraph with Ellipses - As the first sign in the zodiac, the presence
        of Aries always marks the beginning of something energetic and
        turbulent. They are continuously looking for dynamic, speed and
        competition, always being the first in everything - from work to social
        gatherings.
      </Paragraph>
    </div>
  </StorybookComponent>
);

const TypographyStory = () => (
  <React.Fragment>
    <TitleStory />
    <ParagraphStory />
  </React.Fragment>
);

export default TypographyStory;
