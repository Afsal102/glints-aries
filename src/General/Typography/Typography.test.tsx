import * as React from 'react';
import Typography from './Typography';

import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

const { Title, Paragraph } = Typography;
const shortText = 'Glints Aries';

describe('<Typography> should contain the Title and Paragraph components:', () => {
  it(`<Typography.Title> should render a h1 tag with the text Glints Aries`, () => {
    const TitleSnapshot = renderer
      .create(<Title tag="h1">{shortText}</Title>)
      .toJSON();
    expect(TitleSnapshot).toMatchSnapshot();
  });

  it(`<Typography.Paragraph> should render a p tag with the text Glints Aries`, () => {
    const ParagraphSnapshot = renderer
      .create(<Paragraph>{shortText}</Paragraph>)
      .toJSON();
    expect(ParagraphSnapshot).toMatchSnapshot();
  });
});
