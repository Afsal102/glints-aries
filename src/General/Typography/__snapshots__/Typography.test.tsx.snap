// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography> should contain the Title and Paragraph components: <Typography.Paragraph> should render a p tag with the text Glints Aries 1`] = `
<p
  className="aries-typography-paragraph sc-bwzfXH ePzYCu"
  color="#000000"
>
  Glints Aries
</p>
`;

exports[`<Typography> should contain the Title and Paragraph components: <Typography.Title> should render a h1 tag with the text Glints Aries 1`] = `
<h1
  className="aries-typography-title sc-bdVaJa eiimym"
  color="#000000"
>
  Glints Aries
</h1>
`;
